File: src/main/java/main/commands/slashcommands/commands/audio/HistoryCommand.java
package main.commands.slashcommands.commands.audio;

import main.audiohandlers.RobertifyAudioManager;
import main.audiohandlers.TrackScheduler;
import main.commands.prefixcommands.audio.QueueCommand;
import main.utils.GeneralUtils;
import main.utils.RobertifyEmbedUtils;
import main.utils.component.interactions.AbstractSlashCommand;
import main.utils.locale.RobertifyLocaleMessage;
import main.utils.pagination.Pages;
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
import org.jetbrains.annotations.NotNull;

public class HistoryCommand extends AbstractSlashCommand {
    @Override
    protected void buildCommand() {
        setCommand(
                getBuilder()
                        .setName("history")
                        .setDescription("See all the songs that have been played in your current listening session")
                        .build()
        );
    }

    @Override
    public String getHelp() {
        return null;
    }

    @Override
    public void onSlashCommandInteraction(@NotNull SlashCommandInteractionEvent event) {
        if (!checks(event)) return;

        final var guild = event.getGuild();
        final var scheduler = RobertifyAudioManager.getInstance().getMusicManager(guild).getScheduler();
        final var queueHandler = scheduler.getQueueHandler();

        if (queueHandler.isPreviousTracksEmpty()) {
            event.replyEmbeds(RobertifyEmbedUtils.embedMessage(guild, RobertifyLocaleMessage.HistoryMessages.NO_PAST_TRACKS).build())
                    .queue();
        } else {
            final var content = new QueueCommand().getPastTrackContent(guild, queueHandler);
            Pages.paginateMessage(content, 10, event);
        }
    }
}

File: src/main/kotlin/main/commands/slashcommands/audio/HistoryCommand.kt
package main.commands.slashcommands.audio

import main.audiohandlers.RobertifyAudioManager
import main.utils.RobertifyEmbedUtils.Companion.replyEmbed
import main.utils.component.interactions.slashcommand.AbstractSlashCommand
import main.utils.component.interactions.slashcommand.models.SlashCommand
import main.utils.locale.messages.HistoryMessages
import main.utils.pagination.PaginationHandler
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent

class HistoryCommand : AbstractSlashCommand(
    SlashCommand(
        name = "history",
        description = "See all the songs that have been played in your current listening session"
    )
) {

    override suspend fun handle(event: SlashCommandInteractionEvent) {
        val guild = event.guild!!
        val scheduler = RobertifyAudioManager[guild].scheduler
        val queueHandler = scheduler.queueHandler

        if (queueHandler.isPreviousTracksEmpty)
            event.replyEmbed { embed(HistoryMessages.NO_PAST_TRACKS) }
                .queue()
        else {
            val content = QueueCommand().getPastContent(guild, queueHandler).reversed()
            PaginationHandler.paginateMessage(event, content)
        }
    }

}